### PR TITLE
doc: Add a draft version of the SPEC document

### PR DESCRIPTION
**Pull request by Rqnsom**
- The initial version contains general Ziggurat info.
- Documentation for the first two conformance tests.

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+### ZG-CONFORMANCE-001
+
+    The node correctly performs a handshake from the responder side.
+
+    ->
+    -> http handshake request (websocket upgrade)
+    <- http handshake response (websocket upgrade accept)
+
+    Assert: the node’s peer count has increased to 1 and the synthetic node is an established peer.
```
Comment (Rqnsom) [inline SPEC.md:86]:
```
    -> http handshake request (websocket upgrade)
    <- http handshake response (websocket upgrade accept)
```
 I would gladly accept a better solution here 🙂

File: SPEC.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Introduction
+
+The purpose of this index is to provide an overview of the testing approaches to be implemented by Ziggurat. It is intended to evolve as the framework matures, leaving room for novel cases and extensions of existing cases, as called for by any protocol idiosyncrasies that may come to light during the development process.
+
+Some test cases have been consolidated when similar behaviour is tested with differing messages. The final implementation of these cases will be subject to factors such as node setup and teardown details, test run time (and potentially runtime) constraints, readability and maintainability.
+
+## Special Considerations
+
+_To be done_
```
Comment (niklaslong) [inline SPEC.md:9]: I'd suggest removing this section entirely until there is something to document here.

**Review by aphelionz — APPROVED** [review]
LGTM pending Niklas' comment

File: SPEC.md (hunk 1 above)
Comment (niklaslong) [reply SPEC.md:86]: I think this is quite clear, what would you like to improve?

**Review by niklaslong — APPROVED** [review]
LGTM overall 👍

File: SPEC.md (hunk 1 above)
Comment (Rqnsom) [reply SPEC.md:86]: > I think this is quite clear

Then it's all good. I wasn't sure if that was the fanciest way to show an HTTP handshake.

File: SPEC.md (hunk 2 above)
Comment (Rqnsom) [reply SPEC.md]: Done

**Pull request merged**